scripts/simulate.py

```diff
@@ -0,0 +1,192 @@
+    print(f"File exists and clobber=False; exiting.")
+    sys.exit()
+
+if args.verbose:
+    print("Preparing simulation...")
+
+instrument_parameters = {}
+defaults = config.get("defaults", None)
+if defaults is not None:
+    # TODO: update the defaults module to handle validating the choice of defaults
+    if not isinstance(defaults, str):
+        msg = "Defaults must be specified using a string. The string may be either "
+        msg += "a keyword for a supported HERA observing season, or a path to a "
+        msg += "configuration yaml. Refer to the tutorial on hera_sim.defaults for "
+        msg += "details regarding how to interact with the defaults module."
+        raise ValueError(msg)
+
+    print(f"Using default configuration: {defaults}")
+    hera_sim.defaults.set(defaults, refresh=True)
+
+telescope_params = config.get("telescope", {})
+array_layout = telescope_params.get("array_layout", None)
+if array_layout is not None:
+    # TODO: update Simulator to handle having a string passed for array_layout
+    if isinstance(array_layout, str):
+        array_layout = _parse_layout_csv(array_layout)
+    
+    instrument_parameters["array_layout"] = array_layout
+
+# Finish filling out the instrument parameters and initialize the Simulator.
+frequency_parameters = config.get("freq", {})
+instrument_parameters.update(frequency_parameters)
+time_parameters = config.get("time", {})
+instrument_parameters.update(time_parameters)
+hera_sim.defaults.set(instrument_parameters, refresh=False)
+sim = hera_sim.Simulator()
+
+# next: prepare Simulator object, sky, and beam
+# Now handle the extras.
+if telescope_params.get("omega_p", None) is not None:
+    hera_sim.defaults.set({"omega_p": telescope_params["omega_p"]}, refresh=False)
+
+sky_parameters = config.get("sky", {})
+Tsky_mdl = sky_parameters.pop("Tsky_mdl", None)
+if Tsky_mdl is not None:
+    hera_sim.defaults.set({"Tsky_mdl": Tsky_mdl}, refresh=False)
+
+systematics_parameters = config.get("systematics", {})
+default_components = ["foregrounds", "eor", "noise", "rfi", "sigchain"]
+simulation_info = config.get(
+    "simulation", 
+    {"components": default_components, "exclude": []}
+)
+all_simulation_parameters = {
+    component: parameter_dict 
+    for parameters in (sky_parameters, systematics_parameters)
+    for component, parameter_dict in parameters.items()
+}
+simulation_parameters = {}
+for component in simulation_info["components"]:
+    parameters = all_simulation_parameters.get(component, {})
+    for this_component, settings in parameters.items():
+        if this_component in simulation_info.get("exclude", []):
+            continue
+
+        simulation_parameters[this_component] = settings
+
+if args.verbose:
+    print("Running simulation...")
+
+# XXX it may be simpler to just use sim.run_sim for this, and if the user 
+# wants to save the components, then we can loop over the items in 
+# simulation_parameters and just do sim.get
```

Comment:
Not sure I understand this comment, but thought I should note it here.

Reply:
I think I'm just going to scrap the comment. The `get` method re-runs the simulation step, so it would make the CLI significantly slower when running the CLI with the `--save_all` flag.